### ISSUE TITLE
Reboot Loop If Network Blocks Steam

### ISSUE THREAD

**Issue by tiberiusferreira** [ValveSoftware/SteamOS#558]
#### Your system information

* Steam client version: The one bundled with SteamOS 2.87
* SteamOS version: SteamOS 2.87
* Opted into Steam client beta?: [Yes/No] No
* Opted into SteamOS beta?: [Yes/No] No
* Have you checked for updates in Settings > System?: [Yes/No] No internet access

#### Please describe your issue in as much detail as possible:
Describe what you _expected_ should happen and what _did_ happen.

I expected to get to a working configuration page so I could set my proxy and internet configuration.

 However when the internet cable is connected and the network requires a proxy or other configuration to allow Steam to update itself the installer thinks it has connection, Steam fails to update and after the reboot and "Getting drivers ready" page it goes through a "steamos is recovering from a fatal error" boot loop which you can't get out of. 

#### Steps for reproducing this issue:

1. Block Steam from your network (think college internet which blocks Steam or requires some special configuration such as proxies).
2. Try the automated install using USB stick.
3. Get trapped in a "steamos is recovering from a fatal error" loop.

**Comment by ProfessorKaos64** [comment]
Steam has always required the presence of a network, since it runs the updater at the start. As far as I can remember.

**Comment by tiberiusferreira** [comment]
Yes, it requires. But it continues normally even if the existing network blocks steam packages (as is the case with the one I have access now).

**Comment by Lukas-Batema** [comment]
Bump! I am having a similar - or same issue - on my iMac (I purposefully deleted macOS), where it won't connect to the wireless modem. The error says:
```txt
Fatal Error: Steam needs to be online to update. Please confirm your network connection and try again
```
Below that it has an non-clickable "OK" button that has a 25s countdown to "reboot", which starts the process over again. Even when inserting the Ethernet cord, it refuses to work. I cannot use a USB stick to install another OS, or anything. I'm just presented with this, eternally. Any help would be appreciated!
